alpha/action/render.go

```diff
@@ -454,3 +482,150 @@ func combineConfigs(cfgs []declcfg.DeclarativeConfig) *declcfg.DeclarativeConfig
 	}
 	return out
 }
+
+func isBundle(entries []os.DirEntry) bool {
+	foundManifests := false
+	foundMetadata := false
+	for _, e := range entries {
+		if e.IsDir() {
+			switch e.Name() {
+			case "manifests":
+				foundManifests = true
+			case "metadata":
+				foundMetadata = true
+			}
+		}
+		if foundMetadata && foundManifests {
+			return true
+		}
+	}
+	return false
+}
+
+func isPackageManifest(entries []os.DirEntry) bool {
+	for _, e := range entries {
+		if strings.HasSuffix(e.Name(), ".package.yaml") || strings.HasSuffix(e.Name(), ".package.yml") {
+			return true
+		}
+	}
+	return false
+}
+
+type imageReferenceTemplateData struct {
+	Package string
+	Name    string
+	Version string
+}
+
+func (r *Render) renderPackageManifest(ctx context.Context, ref string) (*declcfg.DeclarativeConfig, error) {
+	db, err := sql.Open("sqlite3", fmt.Sprintf("file:%s?mode=memory&cache=shared&_foreign_keys=on", ref))
```

Comment:
I'm sure I'm missing some context here, but I'm wondering...  why we are reading a yaml file into an sqlite DB? 
For e.g. an example I find for this is something like
```yaml
channels:
- currentCSV: myoperator.v0.1.0
  name: original
- currentCSV: myoperator.v1.9.0
  name: alpha
defaultChannel: alpha
packageName: myop
```

Additional question:  
is doing this going to make fulfilling https://github.com/operator-framework/operator-registry/issues/1105 more difficult?

Reply:
I removed this code from the PR because I don't think we need to support package manifest directories anymore.

But to answer your question, this code would take a package manifests root directory (not just the packages.yml file) and load it into an sqlite DB. The package manifest format was actually declarative in that it put all of the bundle manifests for all of the versions in a particular directory structure, which meant that one could idempotently rebuild the database by tweaking something in the source files and constructing the DB again from scratch.